src/main/scala/abalone/variant/Variant.scala

```diff
@@ -53,27 +54,169 @@ abstract class Variant private[variant] (
+          .filterNot(situation.board.pieces.contains(_)) // ...keep the piece only if the landing square is empty
+          .map(landingSquare => 
+            Move(piece, pos, landingSquare, situation, boardAfter(situation, pos, landingSquare), false)
+          )
+          // instead of .filterNot etc, if we wanted to take care of several cases directly from here, we could have written :
+          // .flatMap {
+          //   case (landingSquare) if (!situation.board.pieces.contains(landingSquare)) =>
+          //       Some(Move(piece, pos, landingSquare, situation, boardAfter(situation, pos, landingSquare), false))
+          //   case (landingSquare) if (another condition)
+          //   case _ => None
+          // }
+        )
+    }.toMap
+
+  def validSideMoves(@nowarn situation: Situation):  Map[Pos, List[Move]] = {
+    Map()
+  }
+
+  def validLineMoves(@nowarn situation: Situation):  Map[Pos, List[Move]] = {
+    Map()
+    // situation.board.piecesOf(situation.player).flatMap {
+    //   case ((pos, piece)) => {
+    //     Map(pos -> List({
+    //       Stone.dirs.flatMap { // if the piece would remain on the board after applying a direction...
+    //         case dir => (dir, dir(pos))
+    //         case _ => None
+    //       }.flatMap { // ...check that it would move to square occupied by one of our pieces
+    //         case ((dir: Direction, landingSquare: Option[Pos]))
+    //           if (situation.board.pieces.contains(landingSquare) && situation.board.pieces(landingSquare).player == situation.player) => 
+    //             // let's check if the next square is occupied by us too, or empty :
+    //             if(dir(landingSquare))
+    //               Some(Move(piece, pos, landingSquare, situation, boardAfter(situation, pos, landingSquare), false))
+    //             else
+    //               None
+            
+    //         case _ => None
+    //       }
+    //     }).flatten)
+    //   }
+    // }.toMap
+  }
+
+  def validPushes(@nowarn situation: Situation):  Map[Pos, List[Move]] = {
+    Map()
+  }
+
+  def boardAfter(situation: Situation, orig: Pos, dest: Pos): Board = {
+    // 1. move pieces and get the score
+    val (pieces, capture)   = piecesAfterAction(situation.board.pieces, situation.player, orig, dest)
+
+    // 2. update the board
+    situation.board.copy(
+      pieces = pieces,
+      history = History(
+        situation.history.currentTurn,
+        situation.history.currentTurn, // @TODO: fix this, understand how to create the new Uci with the move being played
+        situation.history.positionHashes,
+        if (capture) situation.history.score.add(situation.player) else situation.history.score,
+        situation.history.halfMoveClock + 1
+      )
+    )
+  }
+/* 
+case class History(
+    lastTurn: List[Uci] = List.empty,
+    currentTurn: List[Uci] = List.empty,
+    positionHashes: PositionHash = Array.empty,
```

Comment:
assume in general these comments will be cleaned after final commit?

Reply:
This one should already have been cleaned as it does not look very good even if the PR targets another branch instead of a main one (and thus another PR will have to be created)
Will proceed to remove these, thanks

Reply:
great